src/utils/ProteomicsLFQ.cpp

```diff
@@ -0,0 +1,1598 @@
+    setValidFormats_("out_cxml", ListUtils::create<String>("consensusXML"));
+
+    registerDoubleOption_("proteinFDR", "<threshold>", 0.05, "Protein FDR threshold (0.05=5%).", false);
+    setMinFloat_("proteinFDR", 0.0);
+    setMaxFloat_("proteinFDR", 1.0);    
+
+    //TODO expose all parameters of the inference algorithms (e.g. aggregation methods etc.)?
+    registerStringOption_("protein_inference", "<option>", "aggregation",
+      "Infer proteins:\n" 
+      "aggregation  = aggregates all peptide scores across a protein (by calculating the maximum) \n"
+      "bayesian     = computes a posterior probability for every protein based on a Bayesian network",
+      false, true);
+    setValidStrings_("protein_inference", ListUtils::create<String>("aggregation,bayesian"));
+
+    registerStringOption_("protein_quantification", "<option>", "unique_peptides",
+      "Quantify proteins based on:\n"
+      "unique_peptides = use peptides mapping to single proteins or a group of indistinguishable proteins"
+      "(according to the set of experimentally identified peptides).\n"
+      "strictly_unique_peptides = use peptides mapping to a unique single protein only.\n"
+      "shared_peptides = use shared peptides only for its best group (by inference score)", false, true);
+    setValidStrings_("protein_quantification", ListUtils::create<String>("unique_peptides,strictly_unique_peptides,shared_peptides"));
+
+
+    registerStringOption_("targeted_only", "<option>", "false", "Only ID based quantification.", false, true);
+    setValidStrings_("targeted_only", ListUtils::create<String>("true,false"));
+
+    registerStringOption_("transfer_ids", "<option>", "false", "Requantification.", false, true);
+    setValidStrings_("transfer_ids", ListUtils::create<String>("false,merged,SVM"));
+
+    registerStringOption_("mass_recalibration", "<option>", "true", "Mass recalibration.", false, true);
+    setValidStrings_("mass_recalibration", ListUtils::create<String>("true,false"));
+
+
+    /// TODO: think about export of quality control files (qcML?)
+
+    Param pp_defaults = PeakPickerHiRes().getDefaults();
+    Param ffi_defaults = FeatureFinderIdentificationAlgorithm().getDefaults();
+    Param ma_defaults = MapAlignmentAlgorithmIdentification().getDefaults();
+    ma_defaults.setValue("max_rt_shift", 0.1);
+    ma_defaults.setValue("use_unassigned_peptides", "false");
+    ma_defaults.setValue("use_feature_rt", "true");
+
+    //Param fl_defaults = FeatureGroupingAlgorithmKD().getDefaults();
+    Param fl_defaults = FeatureGroupingAlgorithmQT().getDefaults();
+
+    Param pq_defaults = PeptideAndProteinQuant().getDefaults();
+    // overwrite algorithm default so we export everything (important for copying back MSstats results)
+    pq_defaults.setValue("include_all", "true"); 
+
+    // combine parameters of the individual algorithms
+    Param combined;
+    combined.insert("Centroiding:", pp_defaults);
```

Comment:
can you skip centroiding? many people would like to use the vendor peak picking

Reply:
Yes it automatically detects if the data is already centroided